geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/serialization/avro/AvroSimpleFeatureEncodingsCache.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.locationtech.geomesa.feature.serialization.avro
+
+import org.apache.avro.io.Encoder
+import org.locationtech.geomesa.feature.serialization.{AbstractWriter, SimpleFeatureEncodingsCache}
+
+/** Concrete [[SimpleFeatureEncodingsCache]] for Avro. */
+object AvroSimpleFeatureEncodingsCache extends SimpleFeatureEncodingsCache[Encoder] {
+
+  override protected val datumWritersFactory: () => AbstractWriter[Encoder] = () => new AvroWriter()
```

Comment:
can probably combine some of these classes into 1 file - it's kind of a fine line in scala - some people go overboard and put everything into 1 file.